src/App.tsx

```diff
@@ -15,48 +17,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
```

Comment:
I think it would be better to separate reversing into different function
Now we are checking the condition of reversing each render, even if the button reverse was not clicked